server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1506,6 +1541,8 @@ public abstract void addFields(
             boolean stored
         );
 
+        public abstract long toSortableLong(Number value);
```

Comment:
Now we have a good reason to have this method :).
Maybe add documentation on this method  to explain why we need to method?